File: leetcode/0242-valid-anagram/solution.py
from collections import Counter

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        s_cnt = Counter(s)

        for ch in t:
            if ch not in s_cnt or s_cnt[ch] == 0:
                return False
            
            s_cnt[ch] -= 1
        
        for k in s_cnt.values():
            if k != 0:
                return False
                     
        return True
        

File: leetcode/0242-valid-anagram/solution.rs
impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
        if s.len() != t.len() {
            return false;
        }

        let mut counter = [0; 26];

        for b in s.bytes() {
            counter[(b - b'a') as usize] += 1;
        }

        for b in t.bytes() {
            let i = (b - b'a') as usize;
            if counter[i] == 0 {
                return false;
            }
            counter[i] -= 1;
        }

        !counter.iter().any(|&v| v != 0)
    }
}

